sdk/scheduler/src/test/java/com/mesosphere/sdk/scheduler/SchedulerApiServerTest.java

```diff
@@ -28,7 +28,7 @@ public void testApiServerReady() throws Exception {
         SchedulerApiServer schedulerApiServer = new SchedulerApiServer(12345, Collections.emptyList(),
                 Duration.ofMillis(TIMEOUT_MILLIS));
         new Thread(schedulerApiServer).start();
-        Thread.sleep(TIMEOUT_MILLIS * 10);
```

Comment:
It does: https://github.com/mesosphere/dcos-commons/blob/master/sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/SchedulerApiServer.java#L56

You can poll on that.